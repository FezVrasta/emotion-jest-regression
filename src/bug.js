import React from 'react';
import styled from '@emotion/styled/macro';

export const Bar = styled.div`
  color: blue;
`;

export const Foo = styled(props => (
  <div {...props}>
    <Bar />
  </div>
))`
  color: red;

  ${Bar} {
      color: yellow;
  }
`;
