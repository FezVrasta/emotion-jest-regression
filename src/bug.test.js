import React from 'react';
import { mount } from 'enzyme';
import { Foo, Bar } from './bug';

it('child style to be read by toHaveStyleRule', () => {
  expect(mount(<Foo />)).toMatchSnapshot();
  expect(mount(<Foo />).find(Bar)).toHaveStyleRule('color', 'yellow');
});
