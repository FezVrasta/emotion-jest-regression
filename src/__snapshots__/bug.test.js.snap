// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`child style to be read by toHaveStyleRule 1`] = `
.emotion-2 {
  color: red;
}

.emotion-2 .emotion-1 {
  color: yellow;
}

.emotion-0 {
  color: blue;
}

<Foo>
  <div
    className="emotion-2 emotion-3"
  >
    <Bar>
      <div
        className="emotion-0 emotion-1"
      />
    </Bar>
  </div>
</Foo>
`;
